src/miner.h

```diff
@@ -160,8 +160,8 @@ class BlockAssembler
     /** Construct a new block template with coinbase to scriptPubKeyIn */
     std::unique_ptr<CBlockTemplate> CreateNewBlock(CChainState& chainstate, const CScript& scriptPubKeyIn);
 
-    static std::optional<int64_t> m_last_block_num_txs;
-    static std::optional<int64_t> m_last_block_weight;
+    inline static std::optional<int64_t> m_last_block_num_txs{};
+    inline static std::optional<int64_t> m_last_block_weight{};
```

Comment:
At first glance, I thought adding inline keyword here might cause separate `m_last_block_num_txs` and `m_last_block_weight` variable instances to exist in different translation units (different variables with the same name `miner.cpp` and `rpc/mining.cpp` that would lead to bugs), but apparently there is new linker magic in c++17 to prevent this!

Reply:
✨ m a g i c ✨

Reply:
Good to know this

Reply:
(more background here: https://github.com/bitcoin/bitcoin/pull/21415#discussion_r592546422)